modules/generator/registry/native_histogram.go

```diff
@@ -179,12 +202,14 @@ func (h *nativeHistogram) newSeries(labelValueCombo *LabelValueCombo, value floa
 }
 
 func (h *nativeHistogram) updateSeries(s *nativeHistogramSeries, value float64, traceID string, multiplier float64) {
+	// Use Prometheus native exemplar handling
+	exemplarObserver := s.promHistogram.(prometheus.ExemplarObserver)
+
 	for i := 0.0; i < multiplier; i++ {
-		s.promHistogram.(prometheus.ExemplarObserver).ObserveWithExemplar(
-			value,
-			map[string]string{h.traceIDLabelName: traceID},
-		)
+		// Let Prometheus handle exemplars natively
+		exemplarObserver.ObserveWithExemplar(value, prometheus.Labels{h.traceIDLabelName: traceID})
```

Comment:
I know it was like this already, but should we move the labels out of the loop too?